### PR TITLE
Add Discord Config Best Practice Guide

### PR DESCRIPTION
**Pull request by TheBrambleShark**
This document describes the policies LuzFaltex uses to build Discord-based communities, from a technical perspective.

### CODE REVIEW

**Comment by TheBrambleShark** [comment]
Needs a note regarding granting permissions to users in channel permission overrides and that doing such breaks RBAC specifications.

File: docs/articles/discord-guild.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+This guide is provided by LuzFaltex to outline the way we configure our Discord Guilds (servers) and the reasons behind them. This guide assumes you will be reading this because you have an interest in building a server to our standards, or because you are a LuzFaltex employee tasked with building a new community.
+
+## Initial Setup
+
+We will start out assuming you have already created your guild, chosen its name, and are ready to begin setup. Many of the practices and the reasons behind them are adapted from Microsoft Best Practice for Windows Active Directory. We additionally rely on [NIST INCITS 359-2012](https://standards.incits.org/apps/group_public/project/details.php?project_id=1658).
+
+Going into this guide, it is a good idea to keep in mind these guidelines:
+
+* **Less is more**: You should make the bare minimum changes as possible. Roles should only be given the bare minimum permissions to allow users with that role to do their job (see the [RBAC Section](#RBAC-Role-Based-Access-Control).) Channel and Channel Category overrides should only differ from role permissions where that channel or channel category differs. For instance, staff-only channels.
```
Comment (Saberfiend) [inline docs/articles/discord-guild.md]: Should change to "You should make the least amount of changes possible"

File: docs/articles/discord-guild.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+* **Inheritance is your friend**: Don't be afraid to grant members more than one role, especially if they do more than one job. In LuzFaltex communities, all staff members regardless of their role are moderators. As such, every staff member gets the Moderator role in addition to the role specific to their team.
```
Comment (Saberfiend) [inline docs/articles/discord-guild.md]: "all staff members - regardless of their role - are moderators."

File: docs/articles/discord-guild.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+* **KISS is powerful**: By making as few changes as possible and following this best practice guide as closely as possible, you enable future administrators to understand the server configuration at a glance. Additionally, servers that are simple scale easily, so configurations made when the server was small do not impede growth.
```
Comment (Saberfiend) [inline docs/articles/discord-guild.md]: "By making only necessary changes, and following this best practice guide as closely as possible, ..."

File: docs/articles/discord-guild.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+### RBAC (Role Based Access Control)
+
+> [!IMPORTANT]
+> Groups should only be given the bare minimum permissions to fulfill their function.
```
Comment (Saberfiend) [inline docs/articles/discord-guild.md]: "Groups should only be given the necessary permissions to fulfill their function."

File: docs/articles/discord-guild.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+Our guild configuration depends heavily on RBAC, or Role Based Access Control. RBAC defines what a role is and how it should be utilized.
```
Comment (Saberfiend) [inline docs/articles/discord-guild.md:28]: "...what a role is, and how it..."

File: docs/articles/discord-guild.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+> RBAC enables you to more closely align the roles you assign users and administrators to the actual roles they old within your organization. -- Microsoft Best Practice
```
Comment (Saberfiend) [inline docs/articles/discord-guild.md]: *hold, not old

File: docs/articles/discord-guild.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+As such, the roles in this guide will be created in such a way to match with their definition. Users that have more than one job will be be given each of the appropriate roles, their cumulative permission grants giving them permission to complete their assigned tasks.
```
Comment (Saberfiend) [inline docs/articles/discord-guild.md]: "... their cumulative permissions grant them the ability to complete their assigned tasks."

File: docs/articles/discord-guild.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+Cosmetic roles are roles which are used for changing name colors, sorting users in the user list, or providing meta-info about the player. These roles will be configured identically to the `Everyone` role and will not be granted any additional permissions.
```
Comment (Saberfiend) [inline docs/articles/discord-guild.md]: "... configured identically to the `Everyone` role, and will not be granted..."

File: docs/articles/discord-guild.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+For ease in configuration, all channels should be placed in a channel group, called a Category. Channels in this category will inherit permissions from the category, meaning you are able to configure one item instead of `n` items, where `n` represents the number of channels in the category.
+
+Category permissions should be modified sparingly. In other words, the only permission that should be changed is the one that is different for that channel. For instance, the Everyone group should not be able to see channels in that category but members of the Moderator group are.
+
+While it is possible to override permissions per-channel, this is not recommended as it breaks inheritance. In other words, once a particular channel in a group has a modification made to it, changes made to the category are no longer reflected within that channel. Channel configuration items, like slow mode or marking a channel as NSFW do not affect inheritance.
```
Comment (Saberfiend) [inline docs/articles/discord-guild.md]: "...or marking a channel as NSFW, do not affect..."

**Review by Saberfiend — CHANGES_REQUESTED** [review]
Let me know if you have any questions!

File: docs/articles/discord-guild.md (hunk 2 above)
Comment (TheBrambleShark) [reply docs/articles/discord-guild.md]: Done, but used commas instead.

File: docs/articles/discord-guild.md (hunk 5 above)
Comment (TheBrambleShark) [reply docs/articles/discord-guild.md:28]: Comma not necessary here.

File: docs/articles/discord-guild.md (hunk 8 above)
Comment (TheBrambleShark) [reply docs/articles/discord-guild.md]: Comma not necessary here.

File: docs/articles/discord-guild.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+This guide is provided by LuzFaltex to outline the way we configure our Discord Guilds (servers) and the reasons behind them. This guide assumes you will be reading this because you have an interest in building a server to our standards, or because you are a LuzFaltex employee tasked with building a new community.
+
+## Initial Setup
+
+We will start out assuming you have already created your guild, chosen its name, and are ready to begin setup. Many of the practices and the reasons behind them are adapted from Microsoft Best Practice for Windows Active Directory. We additionally rely on [NIST INCITS 359-2012](https://standards.incits.org/apps/group_public/project/details.php?project_id=1658).
+
+Going into this guide, it is a good idea to keep in mind these guidelines:
+
+* **Less is more**: You should make the least amount changes as possible. Roles should only be given the bare minimum permissions to allow users with that role to do their job (see the [RBAC Section](#RBAC-Role-Based-Access-Control).) Channel and Channel Category overrides should only differ from role permissions where that channel or channel category differs. For instance, staff-only channels.
```
Comment (Saberfiend) [inline docs/articles/discord-guild.md]: "least amount of changes possible"
Just take out "as" and add "of"
Other than that, everything's good :P

**Review by Saberfiend — CHANGES_REQUESTED** [review]

**Review by Saberfiend — APPROVED** [review]
Looks good to me!

**Pull request merged**